src/send_and_confirm.rs

```diff
@@ -63,7 +63,15 @@ impl Miner {
 
         // Set compute unit price
         final_ixs.push(ComputeBudgetInstruction::set_compute_unit_price(
-            self.priority_fee.unwrap_or(0),
+            if self.dynamic_fee {
+                if let Some(dynamic_fee) = self.dynamic_fee().await {
+                    dynamic_fee
+                }else{
+                    self.priority_fee.unwrap_or(0)
+                }
+            } else {
+                self.priority_fee.unwrap_or(0)
+            }        
```

Comment:
If dynamic fee is requested, we set it the loop. We intentionally skipped fetching priority fee here because you'll fetch it again on the first time through the submit loop below.